### DISCUSSION TITLE
WizardMac/ReadStat as an Extension

### DISCUSSION THREAD

**Discussion by emraher** [duckdb/duckdb#14504] [Ideas]
[tidyverse/haven](https://github.com/tidyverse/haven) uses [WizardMac/ReadStat](https://github.com/WizardMac/ReadStat) to read and write various data formats used by other statistical packages. Is it possible to integrate ReadStat as an extension to duckdb?

**Comment by carlopi** [comment]
I haven't looked at this properly, but this is definitively possible.

Main path for this would be as an community extension, where some interested party can put this together and share to users.

**Reply by emraher** [reply]
I'm really interested  in this but I believe I'm not capable of doing it.  Maybe @hadley or @krlmlr can do something about it.

**Comment by dylanmeysmans** [comment]
> [tidyverse/haven](https://github.com/tidyverse/haven) uses [WizardMac/ReadStat](https://github.com/WizardMac/ReadStat) to read and write various data formats used by other statistical packages. Is it possible to integrate ReadStat as an extension to duckdb?

@carlopi @emraher Including this for people reading this discussion, but not aware of the latest developments yet: this is now available as https://duckdb.org/community_extensions/extensions/read_stat.html.

**Reply by krlmlr** [reply]
This is awesome, thanks for sharing! Looking forward to playing with it and to checking how this works in duckplyr!
